.github/workflows/ci-pipeline.yaml

```diff
@@ -31,6 +31,9 @@ jobs:
       - name: Check types
         run: yarn tsc --noEmit
 
+      - name: Build package
+        run: yarn build
```

Comment:
Is this going to fail under all cases that the `yarn tsc --noEmit` will?

Reply:
no it's slightly different
yarn build builds both our cjs and esm side whereas yarn tsc just checks the types work

Reply:
Yeah, but I guess the question is whether the build checks the types as well.

Reply:
yes it will

Reply:
which I guess makes the check types step superfluous as build will do more

Reply:
That's exactly what I was thinking - checking the types is exactly what we want, but we don't want to do it twice unless the timings need work.